packages/core/shared/stream-parts.ts

```diff
@@ -48,9 +48,50 @@ const functionCallStreamPart: StreamPart<
     };
   },
 };
-
-const dataStreamPart: StreamPart<'2', 'data', Array<JSONValue>> = {
+const toolCallStreamPart: StreamPart<
+  '2',
```

Comment:
@Archimagus the tests are currently broken for me when I run `pnpm test` inside `packages/core` on your branch. Reverting the numbering and using a new number for tool calls would probably fix most of the tests. Please let me know when the tests pass and I'll take another look.

Reply:
I think the tests were also broken on main when I merged last,  I'll check again.  And sure I can use a new number for tools.

Reply:
So, it all seems to be there, the the UI tests all hang forever.  I am looking, but as of right now I'm kind of stumped.  If I skip all the UI tests then everything is fine, but enabling any of them and it just hangs forever.  I can't even get a console log to print from anywhere in a test.

Reply:
I just checked out your PR and all tests (incl. UI) are passing for me locally (at least with `pnpm test` inside `packages/core`).

Reply:
Well that's infuriating.  The UI tests never end on my machine, I see all the tests pass for both edge and node, and then it tries to start the UI and it just spins.  Same thing on the github, the tests are failing because they time out after 30 minutes.  (They don't even obey timeouts specified on the tests) It's like it can't initialize something.  I've tried pnpm install, pnpm rebuild,  deleting my node_modules folder, so far I can't figure out what is causing them to hang.

Reply:
Btw, I'm working on switching the tests over to Vite, 🤞🏻 that this solves a bunch of issues if it works.

Reply:
I'll check my Node version when I get home.  But that still doesn't explain why it fails on the github PR.  Either way,  yeah Vite test would be amazing.

Reply:
heads-up: the vitest PR has landed. maybe that helps with your testing issues